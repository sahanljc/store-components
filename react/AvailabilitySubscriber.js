export { default } from './components/AvailabilitySubscriber'
