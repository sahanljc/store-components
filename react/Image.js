import Image from './components/Image/index'

export default Image
