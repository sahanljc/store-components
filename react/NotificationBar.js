import NotificationBar from './components/NotificationBar/index'

export default NotificationBar
