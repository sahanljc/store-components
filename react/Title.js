import Title from './components/Title/index'

export default Title
