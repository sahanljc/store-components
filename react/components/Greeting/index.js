import PropTypes from 'prop-types'
import React, { Fragment } from 'react'
import { compose, branch, renderComponent } from 'recompose'
import { FormattedMessage } from 'react-intl'
import { path } from 'ramda'
import { graphql } from 'react-apollo'
import gql from 'graphql-tag'

import Loader from './Loader'

import styles from './styles.css'

export const ORDER_FORM_QUERY = gql`
  query {
    minicart @client {
      orderForm
    }
  }
`

const Wrapper = ({ children }) => (
  <div
    className={`${
      styles.greetingContainer
    } mh4 pv4 t-heading-4 c-on-base nowrap`}
  >
    {children}
  </div>
)

const withWrapper = Component => props => (
  <Wrapper>
    <Component {...props} />
  </Wrapper>
)

const Greeting = ({ orderForm }) => {
  const firstName = path(['clientProfileData', 'firstName'], orderForm)
  if (!firstName) return null

  return (
    <Wrapper>
      <Fragment>
        <span className={styles.message}>
          <FormattedMessage id="greeting" />,
        </span>
        <span className={`${styles.firstName} pl2 b`}>{firstName}</span>
      </Fragment>
    </Wrapper>
  )
}

Greeting.propTypes = {
  orderForm: PropTypes.shape({
    clientProfileData: PropTypes.shape({
      firstName: PropTypes.string,
    }),
  }),
}

const withLinkStateOrderForm = graphql(ORDER_FORM_QUERY, {
  props: ({ data: { minicart } }) => ({
    orderForm: minicart && minicart.orderForm ? JSON.parse(minicart.orderForm) : null,
  }),
})

export default compose(
  withLinkStateOrderForm,
  branch(({ loading }) => loading, renderComponent(withWrapper(Loader)))
)(Greeting)
