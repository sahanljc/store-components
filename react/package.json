{
  "private": true,
  "scripts": {
    "pretest": "yarn",
    "test": "vtex-test-tools test",
    "lint": "eslint --ext js,jsx,ts,tsx ."
  },
  "dependencies": {
    "@material-ui/core": "^3.5.1",
    "@vtex/css-handles": "^1.0.2",
    "apollo-client": "^2.5.1",
    "classnames": "^2.2.5",
    "debounce": "^1.2.0",
    "downshift": "^1.31.14",
    "graphql": "~0.13.2",
    "graphql-tag": "~2.8.0",
    "prop-types": "^15.6.2",
    "ramda": "^0.25.0",
    "react": "^16.3.2",
    "react-apollo": "~2.5.2",
    "react-content-loader": "^3.1.2",
    "react-dom": "^16.8.3",
    "react-html-parser": "^2.0.2",
    "react-id-swiper": "^1.6.8",
    "react-intl": "^2.8.0",
    "react-resize-detector": "^3.0.1",
    "react-share": "^2.1.1",
    "react-slick": "^0.23.1",
    "sanitize-html": "^1.20.0",
    "slick-carousel": "^1.8.1"
  },
  "devDependencies": {
    "@vtex/test-tools": "^0.1.2",
    "apollo-cache-inmemory": "^1.4.3",
    "eslint": "^5.16.0",
    "eslint-config-vtex-react": "^4.0.0",
    "prettier": "^1.16.4",
    "typescript": "^3.4.1"
  },
  "vtexTestTools": {
    "defaultLocale": "en"
  }
}
