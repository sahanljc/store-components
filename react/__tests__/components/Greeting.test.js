import { InMemoryCache } from 'apollo-cache-inmemory'
import React from 'react'
import { render } from '@vtex/test-tools/react'

import Greeting from '../../Greeting'

describe('<Greeting /> component', () => {
  const data = {
    minicart: {
      __typename: 'Minicart',
      orderForm: JSON.stringify({
        __typename: 'OrderFormClient',
        clientProfileData: {
          __typename: 'ClientProfileData',
          firstName: 'name',
        },
      }),
    },
  }

  const cache = new InMemoryCache()
  cache.writeData({ data })

  const renderComponent = customProps => {
    const props = {
      ...customProps,
    }

    return render(<Greeting {...props} />, { graphql: { cache: cache } })
  }

  it('should be rendered', () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toBeTruthy()
  })

  it('should match snapshot without loading', () => {
    const { asFragment } = renderComponent()
    expect(asFragment({ loading: false })).toMatchSnapshot()
  })

  it('should match snapshot with loading', () => {
    const { asFragment } = renderComponent({ loading: true })
    expect(asFragment()).toMatchSnapshot()
  })
})
