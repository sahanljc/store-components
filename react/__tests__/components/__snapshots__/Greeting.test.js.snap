// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Greeting /> component should match snapshot with loading 1`] = `
<DocumentFragment>
  <div
    class="greetingContainer mh4 pv4 t-heading-4 c-on-base nowrap"
  >
    <svg
      class="content-loader-mock"
    >
      <rect
        height="32"
        rx="1"
        ry="1"
        width="300"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`<Greeting /> component should match snapshot without loading 1`] = `
<DocumentFragment>
  <div
    class="greetingContainer mh4 pv4 t-heading-4 c-on-base nowrap"
  >
    <span
      class="message"
    >
      <span>
        Hello
      </span>
      ,
    </span>
    <span
      class="firstName pl2 b"
    >
      name
    </span>
  </div>
</DocumentFragment>
`;
